src/html-imports.js

```diff
@@ -9,718 +9,754 @@
-      } else {
-        imp.addEventListener('load', loadedImport);
-        imp.addEventListener('error', errorLoadingImport);
+
+  /**
+   * @param {!Element} element
+   * @return {boolean}
+   */
+  function isElementLoaded(element) {
+    let isLoaded = false;
+    if (useNative && isImportLink(element) && element.import &&
+      element.import.readyState !== 'loading') {
+      isLoaded = true;
+    } else if (isIE && element.localName === 'style') {
+      // NOTE: IE does not fire "load" event for styles that have already
+      // loaded. This is in violation of the spec, so we try our hardest to
+      // work around it.
+      // If there's not @import in the textContent, assume it has loaded
+      if (element.textContent.indexOf('@import') == -1) {
+        isLoaded = true;
+        // if we have a sheet, we have been parsed
+      } else if (element.sheet) {
+        isLoaded = true;
+        const csr = element.sheet.cssRules;
+        // search the rules for @import's
+        for (let i = 0, l = csr ? csr.length : 0; i < l && isLoaded; i++) {
+          if (csr[i].type === CSSRule.IMPORT_RULE) {
+            // if every @import has resolved, fake the load
+            isLoaded = Boolean(csr[i].styleSheet);
+          }
+        }
       }
+    } else if (element.localName === 'script' && !element.src) {
+      isLoaded = true;
     }
-  } else {
-    checkDone();
+    return isLoaded;
   }
-}
-
-function isImportLoaded(link) {
-  return useNative ? link.__loaded ||
-      (link.import && link.import.readyState !== 'loading') :
-      link.__loaded;
-}
-
-// make `whenReady` work with native HTMLImports
-if (useNative) {
-  new MutationObserver(function(mxns) {
-    for (var i=0, l=mxns.length, m; (i < l) && (m=mxns[i]); i++) {
-      if (m.addedNodes) {
-        handleImports(m.addedNodes);
-      }
+
+  /**
+   * Creates a new document containing resource and normalizes urls accordingly.
+   * @param {string=} resource
+   * @param {string=} url
+   * @return {HTMLElement}
+   */
+  function makeDocument(resource, url) {
+    const content = /** @type {HTMLElement} */
+      (document.createElement('import-content'));
+    content.style.display = 'none';
+    if (url) {
+      content.setAttribute('import-href', url);
+    }
+    if (resource) {
+      content.innerHTML = resource;
     }
-  }).observe(document.head, {childList: true});
 
-  function handleImports(nodes) {
-    for (var i=0, l=nodes.length, n; (i<l) && (n=nodes[i]); i++) {
-      if (isImport(n)) {
-        handleImport(n);
-      }
+    // Support <base> in imported docs. Resolve url and remove it from the parent.
+    const baseEl = /** @type {HTMLBaseElement} */ (content.querySelector('base'));
+    if (baseEl) {
+      url = Path._resolveUrl(baseEl.getAttribute('href'), url);
+      baseEl.parentNode.removeChild(baseEl);
     }
+    // TODO(sorvell): this is specific to users of <dom-module> (Polymer).
+    fixDomModules(content, url);
```

Comment:
Let's see if we can remove this. Instead HTMLImports can add a custom api `importForElement`.